tests/cpp/common/test_random.cc

```diff
@@ -45,7 +46,23 @@ TEST(ColumnSampler, Test) {
   // Should always be a minimum of one feature
   cs.Init(n, 1e-16f, 1e-16f, 1e-16f);
   ASSERT_EQ(cs.GetFeatureSet(0)->Size(), 1);
+}
 
+// Test if different threads using the same seed produce the same result
+TEST(ColumnSampler, ThreadSynchronisation) {
+  const int64_t num_threads = 10;
+  int seed = 7;
+  int n = 128;
+  std::vector<std::vector<int>> results(num_threads);
+#pragma omp parallel for schedule(static, 1)
+  for (int64_t i = 0; i < num_threads; ++i) {
+    ColumnSampler cs(seed);
+    cs.Init(n, 0.5f, 0.5f, 0.5f);
+    results.at(i) = cs.GetFeatureSet(0)->ConstHostVector();
```

Comment:
Consider testing several levels for several iterations here.

Reply:
Done